### ISSUE TITLE
Cannot use in Python3

### ISSUE THREAD

**Issue by kyuupichan** [keepkey/python-keepkey#3]
$ python3
Python 3.5.2 (default, Sep  3 2016, 11:46:47) 
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.

> > > import keepkeylib.transport_hid
> > > Traceback (most recent call last):
> > >   File "<stdin>", line 1, in <module>
> > >   File "/usr/local/lib/python3.5/site-packages/keepkeylib/transport_hid.py", line 5, in <module>
> > >     from .transport import Transport, ConnectionError
> > >   File "/usr/local/lib/python3.5/site-packages/keepkeylib/transport.py", line 2, in <module>
> > >     from . import mapping
> > >   File "/usr/local/lib/python3.5/site-packages/keepkeylib/mapping.py", line 1, in <module>
> > >     from . import messages_pb2 as proto
> > >   File "/usr/local/lib/python3.5/site-packages/keepkeylib/messages_pb2.py", line 12, in <module>
> > >     import types_pb2
> > > ImportError: No module named 'types_pb2'

The main problem is the generated messages_pb2 is apparently generated with google protobuf for Python2.   If you fix the import with "from . import" you then run into unicode errors.

Probably best to create two messages_pb2 and conditionally import based on the python version?

**Comment by dstanchfield** [comment]
I have gone down to path of trying to make it compatible earlier this year, but protobuf 2 generated code always gave my issues because it was not python 3 compatible, and protobuf 3 had not been released.  But that is no longer the case so I'll take a look.

**Comment by kyuupichan** [comment]
I have something working here with protobuf 3 and python 3. I had to change
the import line in 2 or 3 places from absolute to  "from .  import" but
that was all. I've not tested to what extent it works in Python 2 yet.

Neil

**Comment by kyuupichan** [comment]
I've just tested the 3 pb2 files by copying them from python3 library directory to python2.  They appear to work fine, but I have not tested in depth.  So in summary:

-  protobuf3 seems to generate files comatible with python 2
- by tweaking 2 or 3 imports from e.g. "import types_pb2" to "from . import types_pb2" the files appear to work on both Python 2 and 3

I wonder if the relative import fix is not known to the protobuf guys, or if I was doing something wrong generating the files.

**Comment by kyuupichan** [comment]
By the way, happy to provide the files if you feel they would be useful to you.

**Comment by bgok** [comment]
Cool! Can you do a pull request and we will have a look.

Thanks!

**Comment by kyuupichan** [comment]
Given these are generated files I didn't see much point in those alone.  So I hacked sed to do the right thing in build_pb.sh.  And included the generated files in the pull request.

**Comment by dekoza** [comment]
I'm trying to add Python 3 support but I keep failing. After getting the imports straight and removing unnecessary `unicode("", "utf-8")` which should be simply `u""`, I can't get the tests to work (yeah, I even introduced missing `__init__.py`). Right now I get this error:
```
____________________________________________________________ ERROR collecting tests/test_zerosig.py _____________________________________________________________
../../../.virtualenvs/trezor/lib/python3.5/site-packages/google/protobuf/internal/python_message.py:1087: in MergeFromString
    if self._InternalParse(serialized, 0, length) != length:
../../../.virtualenvs/trezor/lib/python3.5/site-packages/google/protobuf/internal/python_message.py:1109: in InternalParse
    (tag_bytes, new_pos) = local_ReadTag(buffer, pos)
../../../.virtualenvs/trezor/lib/python3.5/site-packages/google/protobuf/internal/decoder.py:181: in ReadTag
    while six.indexbytes(buffer, pos) & 0x80:
E   TypeError: unsupported operand type(s) for &: 'str' and 'int'

During handling of the above exception, another exception occurred:
tests/test_zerosig.py:4: in <module>
    from . import common
tests/common.py:4: in <module>
    from . import config
tests/config.py:5: in <module>
    from keepkeylib.transport_pipe import PipeTransport
keepkeylib/transport_pipe.py:4: in <module>
    from .transport import Transport
keepkeylib/transport.py:2: in <module>
    from . import mapping
keepkeylib/mapping.py:1: in <module>
    from . import messages_pb2 as proto
keepkeylib/messages_pb2.py:29: in <module>
    options=_descriptor._ParseOptions(descriptor_pb2.EnumValueOptions(), '\220\265\030\001'),
../../../.virtualenvs/trezor/lib/python3.5/site-packages/google/protobuf/descriptor.py:869: in _ParseOptions
    message.ParseFromString(string)
../../../.virtualenvs/trezor/lib/python3.5/site-packages/google/protobuf/message.py:185: in ParseFromString
    self.MergeFromString(serialized)
../../../.virtualenvs/trezor/lib/python3.5/site-packages/google/protobuf/internal/python_message.py:1093: in MergeFromString
    raise message_mod.DecodeError('Truncated message.')
E   google.protobuf.message.DecodeError: Truncated message.
```
(from `pytest`)

**Comment by ngmiller** [comment]
Resolved in #13
